app/main.py

```diff
@@ -1,3 +1,20 @@
+import math
+
+
 class OnlineCourse:
-    # write your code here
-    pass
+
+    def __init__(self, name: str, description: str, weeks: int):
+        self.name = name
+        self.description = description
+        self.weeks = weeks
+
+    @classmethod
+    def days_to_weeks(cls, days):
+        return math.ceil(days / 7)
+
+    @classmethod
+    def from_dict(cls, course_dict: dict):
+        cls.name = course_dict["name"]
+        cls.description = course_dict["description"]
+        cls.weeks = cls.days_to_weeks(course_dict["days"])
```

Comment:
It would be better to use `OnlineCourse.days_to_weeks` here. You will learn about this case in the future videos on OOP